doc/under_sampling.rst

```diff
@@ -0,0 +1,388 @@
+:class:`EditedNearestNeighbours` by repeating the algorithm multiple times.
+Generally, repeating the algorithm will delete more data::
+
+   >>> from imblearn.under_sampling import RepeatedEditedNearestNeighbours
+   >>> renn = RepeatedEditedNearestNeighbours(random_state=0)
+   >>> X_resampled, y_resampled = renn.fit_sample(X, y)
+   >>> print(Counter(y_resampled))
+   Counter({2: 4551, 1: 208, 0: 64})
+
+:class:`AllKNN` differs from the previous
+:class:`RepeatedEditedNearestNeighbours` since the number of neighbors of the
+internal nearest neighbors algorithm is increased at each iteration::
+
+  >>> from imblearn.under_sampling import AllKNN
+  >>> allknn = AllKNN(random_state=0)
+  >>> X_resampled, y_resampled = allknn.fit_sample(X, y)
+  >>> print(Counter(y_resampled))
+  Counter({2: 4601, 1: 220, 0: 64})
+
+In the example below, it can be seen that the three algorithms have similar
+impact by cleaning noisy samples next to the boundaries of the classes.
+
+.. image:: ./auto_examples/under-sampling/images/sphx_glr_plot_comparison_under_sampling_004.png
+   :target: ./auto_examples/under-sampling/plot_comparison_under_sampling.html
+   :scale: 60
+   :align: center
+
+See
+:ref:`sphx_glr_auto_examples_pipeline_plot_pipeline_classification.py`,
+:ref:`sphx_glr_auto_examples_under-sampling_plot_comparison_under_sampling.py`,
+and :ref:`sphx_glr_auto_examples_under-sampling_plot_enn_renn_allknn.py`.
+
+.. _condensed_nearest_neighbors:
+
+Condensed nearest neighbors and derived algorithms
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:class:`CondensedNearestNeighbour` uses a 1 nearest neighbor rule to
+iteratively decide if a sample should be removed or not. The algorithm is
+running as followed:
+
+1. Get all minority samples in a set :math:`C`.
+2. Add a sample from the targeted class (class to be under-sampled) in
+   :math:`C` and all other samples of this class in a set :math:`S`.
+3. Go through the set :math:`S`, sample by sample, and classify each sample
+   using a 1 nearest neighbor rule.
+4. If the sample is misclassified, add it to :math:`C`, otherwise do nothing.
+5. Reiterate on :math:`S` until there is no samples to be added.
+
+The :class:`CondensedNearestNeighbour` can be used in the following manner::
+
+  >>> from imblearn.under_sampling import CondensedNearestNeighbour
+  >>> cnn = CondensedNearestNeighbour(random_state=0)
+  >>> X_resampled, y_resampled = cnn.fit_sample(X, y)
+  >>> print(Counter(y_resampled))
+  Counter({2: 116, 0: 64, 1: 25})
+
+However as illustrated in the figure below, :class:`CondensedNearestNeighbour`
+is sensitive to noise and will add noisy samples.
+
+In the contrary, :class:`OneSidedSelection` will use :class:`TomekLinks` to
+remove noisy samples. In addition, more samples will be kept since it will not
+iterate over the samples of the majority class but all samples which do not
+agree with the 1 nearest neighbor rule will be added at once. The class can be
```

Comment:
Two sentence maybe,  will be added at once at the end of sentence doesnt make sense